### ISSUE TITLE
Support ghc 9.12

### ISSUE THREAD

**Issue by erikd** [haskellari/tree-diff#97]

**Comment by ulysses4ever** [comment]
Yes, please!

**Comment by phadej** [comment]
> Yes, please!

I say it one more time, pinging on this kind of issues is tiresome. I have dozens of packages, and individual pings (or to be fair, opening issues to begin with for packages in dependency closure of `aeson`) are not helpful at all. I'm aware that GHC-9.12.1 just got released. 

The spam simply makes me ignore the issues.

I will update the `CONTRIBUTING.md` to actually forbid opening the issues at all until the GHC release had been around for few months.

**Comment by ulysses4ever** [comment]
Sorry about that 😞 I actually was under impression that you did support people opening issues as opposed to PRs: I can sware I saw such a comment, but of course I can't find it now. Good to know you don't. 

I'm also unsure what policy goes for packages not under your personal account (e.g. haskellari /= phadej). So, having a contributors guide is a good idea. 

Better yet would be to add an issue template that would inform the author about the ban on issues around the latest GHCs... That would be the most direct and hence the most effective way to make sure you don't get these.

**Comment by phadej** [comment]
>  I actually was under impression that you did support people opening issues as opposed to PRs: I can sware I saw such a comment, but of course I can't find it now. Good to know you don't.

The PRs are even less helpful, those I simply ignore. it's too much work to correct them than to do myself; no amount of documentation of what I want to see will make it work - the patches are so simple that one missing part makes them worthless. Either I'll fix them myself (which is the same as doing from scratch) or wait unknown amount of time for the contributor to fix them. 

The idea with opening issues is for people to have a link to reference elsewhere. But I don't think @erikd references them anywhere. For him it might be helpful as he sees all of these issues in his open issues list, but nobody else benefits from these as far as I can tell.  In comparison to e.g. issues like https://github.com/agda/agda/issues/7574 referencing further issues. These kind of downstream issues give some motivation: I see what is blocked on me. Again, I don't see this issue blocking anything (I have an idea, e.g. `cabal-install` tests - but it's my prior knowledge which I might forget, and maybe there's some bigger blocker anyway).

> I'm also unsure what policy goes for packages not under your personal account 

The `CONTRIBUTING.md` is there everywhere. 

> Better yet would be to add an issue template

That's an idea, I have to test that.

---

TL;DR trivial and repetitive issues waste my time.

**Comment by erikd** [comment]
> But I don't think @erikd references them anywhere

I do. They are in an issue in an IOG private Git repo.

**Comment by erikd** [comment]
I last commented on this issue over a month ago. Is there an ETA for this being fixed?

**Comment by phadej** [comment]
> Is there an ETA for this package being updated on Hackage?

No. After 9.12.2 is out. https://discourse.haskell.org/t/psa-correctness-issue-in-ghc-9-12/11204

**Comment by erikd** [comment]
Is `tree-diff` known to trigger this issue? If so, what are the symptoms?

**Comment by phadej** [comment]
I don't know whether the issue is triggered or not. GHC-9.12.1 has an incorrect runtime result bug, compiled programs doesn't crash, they produce wrong results. In particular, GHC-9.12.1 itself might go even more wrong, as it's compiled with itself.

**Comment by erikd** [comment]
Does `ghc-9.12.2` fix this issue? If so, can we get this fixed?
